src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -191,28 +218,52 @@ private void Load()
         {
             if (_isModified)
             {
-                var data = await _cache.GetAsync(_sessionKey);
-                if (_logger.IsEnabled(LogLevel.Information) && data == null)
+                if (_logger.IsEnabled(LogLevel.Information))
                 {
-                    _logger.SessionStarted(_sessionKey, Id);
+                    try
+                    {
+                        var data = await _cache.GetAsync(_sessionKey);
+                        if (data == null)
+                        {
+                            _logger.SessionStarted(_sessionKey, Id);
+                        }
+                    }
+                    catch (Exception exception)
+                    {
+                        _logger.SessionCacheReadException(_sessionKey, exception);
+                        return;
+                    }
                 }
-                _isModified = false;
 
                 var stream = new MemoryStream();
                 Serialize(stream);
-                await _cache.SetAsync(
-                    _sessionKey,
-                    stream.ToArray(),
-                    new DistributedCacheEntryOptions().SetSlidingExpiration(_idleTimeout));
 
-                if (_logger.IsEnabled(LogLevel.Debug))
+                try
+                {
+                    await _cache.SetAsync(
+                        _sessionKey,
+                        stream.ToArray(),
+                        new DistributedCacheEntryOptions().SetSlidingExpiration(_idleTimeout));
+                }
+                catch (Exception exception)
                 {
-                    _logger.SessionStored(_sessionKey, Id, _store.Count);
+                    _logger.SessionCacheWriteException(_sessionKey, Id, exception);
+                    return;
                 }
+
+                _isModified = false;
```

Comment:
put _isModified and the logger inside the try/catch. We want someone to be able to retry.